File: dpo_train/translation_output/py-cpp/HumanEval_4_mean_absolute_deviation.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
float mean_absolute_deviation(std::vector<float> numbers) {
    float sum = 0;
    for (auto x : numbers) {
        sum += x;
    }
    return sum / numbers.size();
}
int main() {
    auto candidate = mean_absolute_deviation;
    assert(candidate((std::vector<float>({(float)1.0f, (float)2.0f}))) == (0.5f));
    assert(candidate((std::vector<float>({(float)1.0f, (float)2.0f, (float)3.0f, (float)4.0f}))) == (1.0f));
    assert(candidate((std::vector<float>({(float)1.0f, (float)2.0f, (float)3.0f, (float)4.0f, (float)5.0f}))) == (1.2f));
}

File: test/before_train/trainset_and_testset/cpp-py/HumanEval_4_mean_absolute_deviation.py
from typing import List

def mean_absolute_deviation(numbers: List[float]) -> float:
    return mean_absolute_deviation(numbers[0:len(numbers)])

def check(candidate):
    assert candidate([1.0, 2.0]) == 0.5
    assert candidate([1.0, 2.0, 3.0, 4.0]) == 1.0
    assert candidate([1.0, 2.0, 3.0, 4.0, 5.0]) == 1.2

def test_check():
    check(mean_absolute_deviation)

test_check()
